xbmc/addons/AudioDecoder.cpp

```diff
@@ -16,51 +16,41 @@
  *  <http://www.gnu.org/licenses/>.
  *
  */
+
 #include "AudioDecoder.h"
+
 #include "music/tags/MusicInfoTag.h"
 #include "music/tags/TagLoaderTagLib.h"
 #include "cores/AudioEngine/Utils/AEUtil.h"
 
 namespace ADDON
 {
 
-std::unique_ptr<CAudioDecoder> CAudioDecoder::FromExtension(CAddonInfo addonInfo, const cp_extension_t* ext)
-{
-  std::string extension = CAddonMgr::GetInstance().GetExtValue(ext->configuration, "@extension");
-  std::string mimetype = CAddonMgr::GetInstance().GetExtValue(ext->configuration, "@mimetype");
-  bool tags = CAddonMgr::GetInstance().GetExtValue(ext->configuration, "@tags") == "true";
-  bool tracks = CAddonMgr::GetInstance().GetExtValue(ext->configuration, "@tracks") == "true";
-  std::string codecName = CAddonMgr::GetInstance().GetExtValue(ext->configuration, "@name");
-  std::string strExt = CAddonMgr::GetInstance().GetExtValue(ext->configuration, "@name") + "stream";
-
-  return std::unique_ptr<CAudioDecoder>(new CAudioDecoder(std::move(addonInfo), std::move(extension),
-      std::move(mimetype), tags, tracks, std::move(codecName), std::move(strExt)));
-}
-
-CAudioDecoder::CAudioDecoder(CAddonInfo addonInfo, std::string extension, std::string mimetype,
-    bool tags, bool tracks, std::string codecName, std::string strExt)
-    : CAddonDll(std::move(addonInfo)), m_extension(extension), m_mimetype(mimetype),
-      m_context(nullptr), m_tags(tags), m_tracks(tracks), m_channel(nullptr)
+CAudioDecoder::CAudioDecoder(const BinaryAddonBasePtr& addonInfo)
+  : IAddonInstanceHandler(ADDON_INSTANCE_AUDIODECODER, addonInfo)
 {
-  m_CodecName = std::move(codecName);
-  m_strExt = std::move(strExt);
-  memset(&m_struct, 0, sizeof(m_struct));
+  m_CodecName = addonInfo->Type(ADDON_AUDIODECODER)->GetValue("@name").asString();
+  m_strExt = m_CodecName + "stream";
```

Comment:
I'm look for it.

Was a little at the test herewith distracted: https://github.com/xbmc/xbmc/pull/12338